warehouse/query-core/src/test/java/datawave/query/jexl/visitors/SetMembershipVisitorTest.java

```diff
@@ -0,0 +1,284 @@
+package datawave.query.jexl.visitors;
+
+import com.google.common.collect.Sets;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.exceptions.DatawaveFatalQueryException;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.test.JexlNodeAssert;
+import org.apache.commons.jexl2.parser.JexlNode;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SetMembershipVisitorTest {
```

Comment:
Excellent test coverage. While not required for approval, I would like to see common code and common test infrastructure extracted to common methods. For example, 
- `testContains(query, fields, expectedContains)`
- `testMembership(query, fields, expectedFields)`
- `testMembershipWithTagging(query, fields, expectedQuery, expectedFields)`

Reply:
I took a slightly different approach than suggested, but I think you will still like it.

Reply:
I like it!